rest/route/service.go

```diff
@@ -132,7 +132,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/projects/{project_id}/copy_redacted_variables").Version(2).Post().Wrap(checkUser, addProject, checkProjectAdmin, editProjectSettings).RouteHandler(makeCopyRedactedVars(sc))
 	app.AddRoute("/projects/{project_id}/events").Version(2).Get().Wrap(checkUser, addProject, checkProjectAdmin, viewProjectSettings).RouteHandler(makeFetchProjectEvents(sc))
 	app.AddRoute("/projects/{project_id}/patches").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makePatchesByProjectRoute(sc))
-	app.AddRoute("/projects/{project_id}/recent_versions").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeFetchProjectVersions(sc))
+	app.AddRoute("/projects/{project_id}/recent_versions").Version(2).Get().Wrap(checkUser, addProject, viewTasks).RouteHandler(makeFetchProjectVersions(sc))
```

Comment:
Although it definitely works to catch the error in the `addProject` middleware it seems to me the root of the issue is that the `viewTasks` middleware returns a 401 both if:
1. a project exists and the user doesn't have permissions and 
2. the project doesn't exist

In either case, we look for a matching scope and don't find one, so [we return a 401](https://github.com/evergreen-ci/gimlet/blob/e71f5a4d64124b8049620bb0902f969e1daecc56/middleware_auth.go#L332).
@john-m-liu pointed out that [here](https://github.com/evergreen-ci/evergreen/blob/master/rest/route/middleware.go#L458) we should be checking whether the project_id corresponds to a project in the db rather than if the string is empty. The same is true for [distro scopes](https://github.com/evergreen-ci/evergreen/blob/1bc18e2b55c47b8e3bcac136c539e42edef45b9b/rest/route/middleware.go#L496).

Reply:
Note that the check for a project existing should be in addition to checking if the project ID is empty, not instead of

Reply:
is there/could there be a project with an empty identifier?

Reply:
it's to handle the scenario that someone tries to access a route without providing a project ID in the URL or query params